### PR TITLE
Refactor G Suite Dialog to use New User List Component

### PR DESCRIPTION
**Pull request by cbauerman**
#### Changes proposed in this Pull Request

* Rename generic Component to `GoogleApps` to `GSuiteUpgrade`
* Completely refactor `GSuiteDialog` as a functional component using the new `GSuiteNewUserList` shared component
* Refactor parents ( `GSuiteUpgrade`, `GSuiteNudge` ) to use new props

#### Testing instructions

1. Open console, run `localStorage.setItem('debug', 'calypso:analytics:*');`, keep console open
2. Add a new site for your test account with a domain and plan, pause before checkout.
3. Purchase site and domain.
4. Confirm you are taken to `/checkout:siteSlug/with-gsuite/:domain/:reciept_id`
5. Note your receipt id
6. Confirm new User List component works, add a couple users and click Add Email/Continue
7. Confirm that all users were added to the cart correctly
8. Using React Dev Tools inspect the `CartItems` Element
9. Inspect the `cart` property, drilling down to `cart.products[0].extra`
10. Confirm the `extra` object has a `receipt_for_domain` property with the same receipt id you notes in step 5.
11. Finish checkout
12. Go to `/domains/add/:siteSlug` for any site with a paid plan
13. Chose a domain ( we will not be purchasing it )
14. Confirm you have been taking to `/domains/add/:domain/google-apps/:siteSlug`
15. Open console for event tracking
16. Click "Skip" and confirm `calypso_checkout_gsuite_upgrade_skip_button_click` event
17. Return to step 14.
18. Add some users and click "Yes, Add Email"
19. Confirm `calypso_checkout_gsuite_upgrade_add_email_button_click` event
20.  Confirm users in cart are correct

### CODE REVIEW

**Review by stephanethomas — CHANGES_REQUESTED** [review]
> Confirm you have been taking to /domains/add/:domain/google-apps/:siteSlug

We should update that path to mention `gsuite` instead of `google-apps` but this can of course happen in another pull request.

I had a few suggestions but otherwise it works fine ... except for one thing: when clicking the `Skip` button at step #16 the checkout page loads entirely. It seems `page.js` doesn't perform the redirect. It doesn't happen for the `Purchase G Suite` button though.

**Comment by cbauerman** [comment]
> when clicking the Skip button at step #16 the checkout page loads entirely. It seems page.js doesn't perform the redirect. It doesn't happen for the Purchase G Suite button though

Weird, let's see what we can do to fix that.

File: client/components/upgrades/gsuite/gsuite-dialog/index.jsx [hunk 1]
```diff
@@ -3,249 +3,140 @@
...
-		onAddGoogleApps: PropTypes.func.isRequired,
-		onClickSkip: PropTypes.func.isRequired,
-		analyticsSection: PropTypes.string,
-		initialGoogleAppsCartItem: PropTypes.object,
-	};
-
-	state = {
-		users: null,
-		validationErrors: null,
+const GSuiteDialog = ( {
```
[the remark this answers is not in the document]
Comment (cbauerman) [reply client/components/upgrades/gsuite/gsuite-dialog/index.jsx]: I forgot it is also not a dialog in Calypso terms 🤦‍♂️ I will get it a better, more descriptive name.

**Comment by cbauerman** [comment]
> > when clicking the Skip button at step #16 the checkout page loads entirely. It seems page.js doesn't perform the redirect. It doesn't happen for the Purchase G Suite button though
> 
> Weird, let's see what we can do to fix that.

Turns out a leading space will mess up a page redirect 🙂

**Comment by stephanethomas** [comment]
> Turns out a leading space will mess up a page redirect

Wow, good catch.

**Review by stephanethomas — APPROVED** [review]
I only had a minor comment, otherwise this looks good and works fine.

File: client/components/upgrades/gsuite/gsuite-upsell-card/style.scss [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	font-weight: 100;
+	margin: 0;
+}
+
+.gsuite-upsell-card__product-logo {
+	background: url( '/calypso/images/upgrades/g-suite-logo.png' ) no-repeat left;
+	background-size: 73px;
+	display: inline-block;
+	height: 19px;
+	text-indent: -999999px;
```
[the remark this answers is not in the document]
Comment (cbauerman) [reply client/components/upgrades/gsuite/gsuite-upsell-card/style.scss]: 😅 Let's take this chance to get rid of it

[the remark this answers is not in the document]
Comment (cbauerman) [reply client/components/upgrades/gsuite/gsuite-upsell-card/style.scss]: actually it definitely still does it's job, we can address it later

**Comment by Stojdza** [comment]
@cbauerman This PR broke [one e2e test](https://circleci.com/gh/Automattic/wp-calypso/340843) - `Sign up for a site on a business paid plan w/ domain name coming in via /create as business flow in CAD currency @parallel`. It's because selector has [changed](https://github.com/Automattic/wp-calypso/pull/34133/files#diff-c1dfd624e934d3390ddb96b1796310c2R16) in this PR and the full e2e suite wasn't started against PR so you couldn't catch a failure. 

Is it possible to return selector value for the "Skip" button to be `gsuite-upsell-card__checkout-button`? If not, we can work on a workaround to pass these tests.

**Comment by cbauerman** [comment]
> This PR broke [one e2e test](https://circleci.com/gh/Automattic/wp-calypso/340843) - `Sign up for a site on a business paid plan w/ domain name coming in via /create as business flow in CAD currency @parallel`.

Darn, I'm sorry! Looking at the PR it looked like those selectors were [already](https://github.com/Automattic/wp-calypso/pull/34133/files?file-filters%5B%5D=.js#diff-ed0d30e59fb045357c60a35660c56244) [hit](https://github.com/Automattic/wp-calypso/pull/34133/files?file-filters%5B%5D=.js#diff-c1dfd624e934d3390ddb96b1796310c2) by a search-and-replace I did ( which is really why I should have been able to notice 🤦‍♂ 🤦‍♂ 🤦‍♂  )

@Stojdza since I already changed the selectors is it ok to change them again to be more idiomatic with the rest of the component? #34276 is what I have in mind.

**Pull request merged**